src/tests/functional/visit_tests.ts

```diff
@@ -70,6 +70,16 @@ export class VisitTests extends TurboDriveTestCase {
     this.assert(await this.changedBody)
   }
 
+  async "test navigation back restores from a preview"() {
+    await this.clickSelector("#same-origin-link")
+    this.assert.notOk(await this.hasSelector("html[data-turbo-preview]"), "does not set data-turbo-preview on a visit")
+
+    await this.goBack()
+
+    this.assert.equal(await this.nextAttributeMutationNamed("document", "data-turbo-preview"), "", "sets [data-turbo-preview] on the <html> element")
```

Comment:
I'm not sure how, but this is causing StaleReference errors that are failing the suite: https://github.com/hotwired/turbo/pull/162/checks?check_run_id=1846330642#step:7:178